profiles/management/commands/gen_realistic_users.py

```diff
@@ -0,0 +1,254 @@
+
+    @classmethod
+    def _get_edx_course_key(cls, data):
+        return data['course_id']
+
+    @classmethod
+    def _fill_in_missing_data(cls, data, user, course_run):
+        data['username'] = user.username
+        return data
+
+
+class EnrollmentDeserializer(CachedModelDeserializer):
+    """
+    CachedEnrollment deserializer
+    """
+
+    model_cls = CachedEnrollment
+    data_key = 'enrollments'
+
+    @classmethod
+    def _get_edx_course_key(cls, data):
+        return data['course_details']['course_id']
+
+    @classmethod
+    def _fill_in_missing_data(cls, data, user, course_run):
+        data['user'] = user.username
+        data['course_details'].update({
+            'course_start': course_run.start_date.isoformat(),
+            'course_end': course_run.end_date.isoformat(),
+            'enrollment_start': course_run.enrollment_start.isoformat(),
+            'enrollment_end': course_run.enrollment_end.isoformat()
+        })
+        return data
+
+
+CACHED_MODEL_DESERIALIZERS = [EnrollmentDeserializer, CertificateDeserializer]
+
+
+def deserialize_user_data(user_data, course_runs):
+    """
+    Deserializes a dict of mixed User/Profile data and returns the newly-inserted User
+    """
+    user = User.objects.create(
+        email=user_data['email'],
+        first_name=user_data['first_name'],
+        last_name=user_data['last_name'],
+        # Set the username to make this User easily identifiable as a 'fake'
+        username=FAKE_USER_USERNAME_PREFIX + user_data['email'].split('@')[0],
+        is_active=True,
+        is_staff=False,
+        is_superuser=False,
+    )
+    # Create new cached edX data records for each type we care about and associate them with a User and CourseRun
+    for cached_model_deserializer in CACHED_MODEL_DESERIALIZERS:
+        if cached_model_deserializer.data_key in user_data:
+            accounted_course_runs = [
+                cached_model_deserializer.deserialize(user, data, course_runs)
+                for data in user_data[cached_model_deserializer.data_key]
+            ]
+            unaccounted_course_runs = set(course_runs) - set(accounted_course_runs)
+        else:
+            unaccounted_course_runs = course_runs
+        # For each course run that didn't have associated cached edX data (for Enrollments, Certificates, etc),
+        # create an 'empty' record (data=None)
+        for course_run in unaccounted_course_runs:
+            cached_model_deserializer.model_cls.objects.create(
+                user=user,
+                course_run=course_run,
+                last_request=CACHED_MODEL_LAST_REQUEST
+            )
+    return user
+
+
+def deserialize_profile_data(profile, user_data):
+    """
+    Deserializes a dict of mixed User/Profile data and creates a new Profile
+    """
+    for k, v in user_data.items():
+        if hasattr(profile, k) and not is_relation_field(profile, k):
+            setattr(profile, k, v)
+    profile.save()
```

Comment:
this looks very similar to the `deserialize_model_data` method above, is there a reason we couldn't do:

```python
deserialize_model_data(profile, user_data)
```

instead?